MerlinAU.sh

```diff
@@ -4797,6 +4811,96 @@ _ShowNodesMenu_()
    printf "${SEPstr}"
 }
 
+_ShowNodesMenuOptions_()
+{
+    while true
+    do
+        _ShowNodesMenu_
+        printf "\nEnter selection:  "
+        read -r nodesChoice
+        echo
+        case $nodesChoice in
+            e|exit) break
+               ;;
+            *) _InvalidMenuSelection_
+               ;;
+        esac
+    done
+}
+
+##---------------------------------------##
+## Added by ExtremeFiretop [2024-May-02] ##
+##---------------------------------------##
+_DownloadChangelogs_()
+{
+    changeLogTag="$(echo "$(nvram get buildno)" | grep -qE "^386[.]" && echo "386" || echo "NG")"
+    if [ "$changeLogTag" = "386" ]
+    then
+        wget -O "$FW_BIN_DIR/Changelog-${changeLogTag}.txt" "${CL_URL_386}"
+    elif [ "$changeLogTag" = "NG" ]
+    then
+        wget -O "$FW_BIN_DIR/Changelog-${changeLogTag}.txt" "${CL_URL_NG}"
+    fi
+    changeLogFile="${FW_BIN_DIR}/Changelog-${changeLogTag}.txt"
+    if [ ! -f "$changeLogFile" ]
+    then
+        Say "Change-log file [$changeLogFile] does NOT exist."
+        _DoCleanUp_ 1 "$keepZIPfile" "$keepWfile"
+    else
+        clear
+        printf "\n${GRNct}Changelog is ready to review!${NOct}\n"
+        printf "\nPress '${REDct}q${NOct}' to quit when finished.\n"
+        _WaitForEnterKey_
+        less "$changeLogFile"
+    fi
+    _DoCleanUp_ 1 "$keepZIPfile" "$keepWfile"
```

Comment:
This lines is causing the following error:

`./MerlinAU.sh: line 5019: keepWfile: parameter not set`

Because variable "`keepWfile`" is not defined anywhere.

Also, I don't fully understand the reason to call the"`_DoCleanUp_`" function here just to delete the Changelog file.

Reply:
@Martinski4GitHub 

Fixed, I was tested with the Gnuton version :) keepWfile exists there.